### DISCUSSION TITLE
Suppress Connection Errors

### DISCUSSION THREAD

**Discussion by berndschneider5** [matteocorti/check_ssl_cert#379] [Q&A] [answered]
Is there an option, that the check will not produce a Critical message, if the host/port is not available? Or maybe an Unknown state instead of critical in this case?

**Comment by matteocorti** [comment, marked as answer]
You can use the ```--ignore-connection-problems``` command line option to return an OK state.

Optionally, you can specify the desired return value.

Examples:

```
$ ./check_ssl_cert --host github.com --port 111 --ignore-connection-problems
SSL_CERT OK: Cannot connect to github.com:111
$ ./check_ssl_cert --host github.com --port 111 --ignore-connection-problems 1
SSL_CERT WARNING: Cannot connect to github.com:111
```

**Reply by berndschneider5** [reply]
ok, sorry. i have not seen this in the documentation. maybe i need new glasses :-)

**Reply by matteocorti** [reply]
No problem. I didn't remember it and was already adding a new option. Only then I noticed that it was already there.

**Reply by berndschneider5** [reply]
now i remember... i already used this option, but there was a problem. if i set this, i get randomly a critical error: "Service check timed out after 60.01 seconds" 
as soon as i remove this option, the check works again. 
Any ideas?

**Reply by matteocorti** [reply]
Strange, can you please post the debugging output? (```--debug```)

**Reply by matteocorti** [reply]
Analysed the debugging output: the problem most likely lies in the usage of ```curl``` to check the connection. I committed a new version (3278d75) which uses netcat. Can you please test?
